### DISCUSSION TITLE
Split aws-lambda-nodejs into construct and bundler, and expose bundler

### DISCUSSION THREAD

**Discussion by nihakue** [aws/aws-cdk#12488] [General]
At the moment, it's very difficult to use aws-lambda-nodejs along with other high level lambda function constructs (such as [Lambda@Edge](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-cloudfront-readme.html#lambdaedge)) because they both subclass lambda.Function. The vast majority of the benefit of using aws-lambda-nodejs is the bundler, so it seems like it should be possible to make it available independent of the construct. @jogold do you have any thoughts on this?

In general, is there a best practice pattern for extending L2 constructs in a way that doesn't make them mutually exclusive with other L2 construct extensions? I'm thinking along the lines of plugins, mixins, composition, that sort of thing.

**Comment by indrora** [comment]
Generally, duplication of the construct and refactoring out the shared code between them would be a preferred mechanism. 

Splitting out the bundler would have other, possibly positive effects as well. How would you define this interface?
